hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/FederationClientInterceptor.java

```diff
@@ -314,7 +314,7 @@ public GetNewApplicationResponse getNewApplication(
     // Try calling the getNewApplication method
     List<SubClusterId> blacklist = new ArrayList<>();
     int activeSubClustersCount = getActiveSubClustersCount();
-    int actualRetryNums = Math.min(activeSubClustersCount, numSubmitRetries) + 1;
```

Comment:
Can you explain why +1 is not needed anymore?

Reply:
For example, we have 1 execution method A, and we want this method to be retried 3 times, that is, if the method A fails to execute the first time, it can be tried 3 more times.

FederationActionRetry, the retry count of this class starts from 0. If our retryNum=3, it will be executed up to 4 times, which can already meet our expectations. We do not need to manually increase 1 to ensure that this method will be executed 4 times.